src/java/org/apache/cassandra/db/memtable/AbstractAllocatorMemtable.java

```diff
@@ -233,8 +236,7 @@ private void flushIfPeriodExpired()
             else
             {
                 // we'll be rescheduled by the constructor of the Memtable.
-                owner.signalFlushRequired(AbstractAllocatorMemtable.this,
-                                          ColumnFamilyStore.FlushReason.MEMTABLE_PERIOD_EXPIRED);
+                signalFlushRequired(ColumnFamilyStore.FlushReason.MEMTABLE_PERIOD_EXPIRED, false);
```

Comment:
This should not go through the memtable, but call the owner directly. Keeping a reference to the memtable in the flush method will slow down its collection, likely drastically.